packages/plugins/src/hardhat/tasks.ts

```diff
@@ -436,10 +514,6 @@ task(TASK_CHUGSPLASH_COMMIT)
       })
       spinner.succeed('Compiled deploy config')
 
-      const ipfs = create({
-        url: args.ipfsUrl || 'https://ipfs.infura.io:5001/api/v0',
-      })
```

Comment:
Infura deprecated their public IPFS gateway ([link](https://docs.infura.io/infura/networks/ipfs/how-to/access-ipfs-content/public-gateway)). Since I couldn't find another public IPFS gateway with the ability to write files, I changed this logic to the following:
* If the user adds the `local` flag to this task, then we get the IPFS hash that *would* be published to IPFS without actually publishing it.
* Alternatively, the user can supply their own IPFS URL like before.
* Lastly, the user can supply an Infura project ID and API key, which are now required to use an Infura node.